### PR TITLE
added v2 of the click tracking

### PR DESCRIPTION
**Pull request by orlandina**
I've checked the usage of the event triggers in GA
other teams use it like that:
category : team or project, e.g. finance or de-de/identity
action: the action performed, e.g. register, login, click
label: is not always used, when used is to differentiate the actions, e.g. action:login, label: b2b or b2c

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -1,75 +1,119 @@
...
-    <button id="myButton">trackIt (gtm click tracking)</button>
+        <as24-tracking type="gtm" action="click" as24-tracking-click-target="#myButton" linkgroup="HackList"
+            linkid="V1 (abundance)"></as24-tracking>
+        <button id="myButton">trackIt (gtm click tracking)</button>
 
-    <!--
+        <!--
+        ut.push(['gtm', 'click_v2',{eventcategory: 'team-project', eventaction: 'feature', eventlabel: 'close'}]);
+        -->
+        <as24-tracking type="gtm" action="click_v2" as24-tracking-click-target="#myNewButton"
```
Comment (mpuppin) [inline docs/index.html]: use "click" instead "click_v2"
use just "category", "action", "label" in the <as24-tracking> element
translate to data layer correctly with the prefix "event"
update documentation

File: src/gtm/index.js [hunk 2]
```diff
@@ -54,7 +54,19 @@ function generateCommonParams(data) {
 
 function trackClick(params) {
     gtm.push(generateCommonParams(params));
-    gtm.push({ event: 'click' });
+    gtm.push({
```
Comment (mpuppin) [inline src/gtm/index.js]: check the params to see if is old click or new and then generate the common params or events

File: src/gtm/index.js [hunk 3]
```diff
@@ -54,7 +54,19 @@ function generateCommonParams(data) {
 
 function trackClick(params) {
     gtm.push(generateCommonParams(params));
-    gtm.push({ event: 'click' });
+    gtm.push({
+        event: 'click'
+    });
+}
+
+function trackClick_v2(params) {
```
Comment (mpuppin) [inline src/gtm/index.js]: see above

File: docs/index.html [hunk 4]
```diff
@@ -1,75 +1,119 @@
...
-    <as24-tracking type="gtm" action="set" classified_customer-i-d="test" notification_-i-d=""></as24-tracking>
+        <as24-tracking type="gtm" action="set" classified_customer-i-d="test" notification_-i-d=""></as24-tracking>
 
-    <!--
+        <!--
     ut.push(['gtm', 'click',{linkgroup:'HackList',linkid:'V1 (abundance): ' + itemId + '|' + abundance}]);
     -->
-    <as24-tracking type="gtm" action="click" as24-tracking-click-target="#myButton" linkgroup="HackList" linkid="V1 (abundance)"></as24-tracking>
-    <button id="myButton">trackIt (gtm click tracking)</button>
+        <as24-tracking type="gtm" action="click" as24-tracking-click-target="#myButton" linkgroup="HackList"
```
[the remark this answers is not in the document]
Comment (orlandina) [reply docs/index.html]: @mpuppin I've added a note about deprecated and updated the Docu. I really wanted your input if action="click_v2" is a good way to go or of it is better to do something like action="click" version="2" - add and check for that version param....

File: docs/index.html (hunk 1 above)
Comment (orlandina) [reply docs/index.html]: action is already reserved, so i will come up with a different name

File: docs/index.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (orlandina) [reply docs/index.html]: ok done

File: src/gtm/index.js (hunk 2 above)
Comment (orlandina) [reply src/gtm/index.js]: changed this your way

File: package.json [hunk 5]
```diff
@@ -49,6 +49,7 @@
     "amp-index-of": "^1.1.0",
     "amp-is-date": "^1.0.1",
     "amp-merge": "^1.0.0",
-    "cookies-js": "^1.2.2"
+    "cookies-js": "^1.2.2",
+    "lodash": "^4.17.15"
```
Comment (inakianduaga) [inline package.json]: is `lodash` used somewhere? (I hope not)

**Comment by inakianduaga** [comment]
If we keep adding functionality to this library, we should start providing proper types for all these custom elements / `ut` object in a proper way now that codebases should use TypeScript. Any plans for that? ideally this repo contains a `index.d.ts` with all the type definitions and a key in package.json so they are automatically picked up by compiler/IDEs

File: package.json (hunk 5 above)
Comment (orlandina) [reply package.json]: no it is not used, removed it

**Comment by mpuppin** [comment]
good catch. it is in our tech debt next week

**Pull request merged**